src/uniprotkb/components/entry/tabs/Publications.tsx

```diff
@@ -24,22 +24,110 @@ import { processUrlTemplate } from '../../../../shared/utils/xrefs';
 import {
   CitationsAPIModel,
   Reference,
+  Source,
 } from '../../../../supporting-data/citations/adapters/citationsConverter';
 import LiteratureCitation from '../../../../supporting-data/citations/components/LiteratureCitation';
 import apiUrls from '../../../config/apiUrls/apiUrls';
 import { getParamsFromURL } from '../../../utils/resultsUtils';
 
+const orcidIDRegExp = /(\d{4}-){3}\d{4}/;
+
+type PublicationSourceProps = {
+  accession: string;
+  source: Source;
+};
+
+export const PublicationSource = ({
+  accession,
+  source,
+}: PublicationSourceProps) => {
+  const databaseInfoMaps = useDatabaseInfoMaps();
+
+  if (source.name.includes('UniProtKB')) {
+    return (
+      <>
+        <EntryTypeIcon entryType={source.name} />
+        {source.name}
+      </>
+    );
+  }
+
+  let uriLink =
+    source && databaseInfoMaps?.databaseToDatabaseInfo[source.name]?.uriLink;
+
+  if (source.name === 'ORCID') {
+    return (
+      <>
+        <EntryTypeIcon entryType={source.name} />
+        {source.name}:{' '}
+        <ExternalLink
+          url={
+            source.id && orcidIDRegExp.test(source.id)
+              ? processUrlTemplate(uriLink, { id: source.id })
+              : null
+          }
+        >
+          {source.id}
+        </ExternalLink>
+        {' ('}
+        <ExternalLink
+          url={`//community.uniprot.org/bbsub/bbsubinfo.html?accession=${accession}`}
+        >
+          see community submission
+        </ExternalLink>
+        ).
+      </>
+    );
+  }
+
+  /* URI link exceptions: */
+  // 'GeneRif' is indexed as 'GeneRIF' in the configuration endpoint
+  if (source.name === 'GeneRif') {
+    uriLink = databaseInfoMaps?.databaseToDatabaseInfo.GeneRIF?.uriLink;
+  }
+  // 'MGI' ID is missing the initial "MGI:"
+  if (source.name === 'MGI') {
+    uriLink = uriLink?.replace('%id', 'MGI:%id');
+  }
+  // 'IEDB' is not in the configuration endpoint
+  if (source.name === 'IEDB') {
+    uriLink = 'https://iedb.org/antigen/UNIPROT:%id';
+  }
+  // 'IMPC' is not in the configuration endpoint
+  if (source.name === 'IMPC') {
+    uriLink = 'https://www.mousephenotype.org/data/genes/%id';
+  }
+
+  // 'GAD' not in the configuration endpoint, but don't inject a link
```

Comment:
Does the `'GAD' not in...` line apply to L104-L110? I was looking for `GAD` to appear somewhere, so it looks like its more general? Maybe something along the lines of "Some DBs are dead and not in the configuration endpoint (eg GAD) and so we only display the accession but not the link."

Reply:
I've changed a bit this section as suggested